grakn-graql/src/main/java/ai/grakn/graql/internal/query/aggregate/MinAggregate.java

```diff
@@ -39,25 +36,29 @@
     }
 
     @Override
-    public Optional<T> apply(Stream<? extends Answer> stream) {
-        return stream.map(this::getValue).min(naturalOrder());
+    public Number apply(Stream<? extends Answer> stream) {
+        NumberPrimitiveTypeComparator comparator = new NumberPrimitiveTypeComparator();
```

Comment:
Ideally the comparator instance would be passed in via constructor, however in this case this may make sense since the comparator implementation is unlikely to need to be swapped out for something else with the same interface in the future. Let's discuss!